templates/manila/bin/healthcheck.py

```diff
@@ -0,0 +1,230 @@
+
+class HeartbeatServer(server.BaseHTTPRequestHandler):
+    @classmethod
+    def initialize_class(cls, binary, config_file):
+        """Calculate and initialize constants"""
+        cls.binary = 'manila-' + binary
+        cls.config_file = config_file
+
+        if binary != 'share':
+            services_filters = [
+                {'host': CONF.host,
+                 'state': STATE_UP,
+                 'status': STATUS_ENABLED,
+                 'binary': cls.binary}]
+        else:
+            backend_opts = [
+                cfg.StrOpt('backend_host'),
+                cfg.StrOpt('backend_availability_zone', default=None),
+            ]
+            services_filters = []
+            for backend in CONF.enabled_share_backends:
+                conf = share_conf.Configuration(backend_opts, backend)
+                host = "%s@%s" % (CONF.host, backend)
+
+                # We also want to set cluster to None on empty strings, and we
+                # ignore leading and trailing spaces.
+                cluster = CONF.cluster and CONF.cluster.strip()
+                cluster = (cluster or None) and f'{cluster}@{backend}'
+
+                az = (conf.backend_availability_zone or
+                      CONF.storage_availability_zone)
+                services_filters.append({'host': host,
+                                         'zone': az,
+                                         'status': STATUS_ENABLED,
+                                         'binary': cls.binary})
+
+        # Store the filters used to find out services.
+        cls.services_filters = services_filters
+
+    @staticmethod
+    def check_service(services, **filters):
+        # Check if services were already created and are registered in the db
+        our_services = [service for service in services
+                        if all(service.get(key, None) == value
+                               for key, value in filters.items())]
+        if not our_services:
+            raise Exception(f'Not found: Service not found with: {filters}')
+
+        num_services = len(our_services)
+        if num_services != 1:
+            print(f'Too many services found! {num_services}')
+
+        ok = all(service['state'] == STATE_UP for service in our_services)
+        if not ok:
+            raise Exception('Service error: Service is not UP')
+
+    def _parse_service_list_output(self, services_list):
+        # Use only some of the fields we know are available in the command 
+        # output. These can change in the future and the change would not
+        # be backwards compatible with this healthcheck, so we keep only 
+        # what is essential.
+        service_headers = ['Binary', 'Host', 'Zone']
+        services_list = services_list.decode('utf-8').split('\n')
+
+        # Alongside the output we want, we have a chance of getting some 
+        # irrelevant things, so we drop that and get the index of where we
+        # should start checking on the output.
```

Comment:
This is an example of what an output of the command I'm running looks like [1]. The first line is basically an info log
https://paste.opendev.org/show/bYU4HEQ8OYAKhx0nIdLO/
Apparently if I use the --use-journal instead of --nodebug, all info and debug will be redirected, so I made that change and avoided